### PR TITLE
Add INLINE pragma to 'with'

### PR DESCRIPTION
**Pull request by jystic**
Some blizzard profiling revealed that this wasn't being inlined.

/cc @amosr @nhibberd

### CODE REVIEW

File: src/P/Functor.hs [hunk 1]
```diff
@@ -8,5 +8,6 @@ import           Data.Function (flip)
 import           Data.Functor as X (Functor(..), ($>), (<$>), void)
 
 with :: Functor f => f a -> (a -> b) -> f b
-with =
-  flip fmap
+with xs f =
+  fmap f xs
+{-# INLINE with #-}
```
Comment (erikd-ambiata) [inline src/P/Functor.hs]: `INLINE` or `INLINABLE` ?

Comment (jystic) [inline src/P/Functor.hs]: `INLINABLE` would be ok, but I can't think of a reason why you wouldn't always want to inline this as it is just a synonym for `fmap`, so inlining will never increase code size

Comment (erikd-ambiata) [inline src/P/Functor.hs]: Yep, makes sense. :+1:

**Comment by nhibberd** [comment]
:+1:

**Pull request merged**